lib/desktop-notifications.ts

```diff
@@ -46,6 +35,25 @@ function closeNotification(id: string) {
   return nativeModule.closeNotification(id)
 }
 
+type DesktopNotificationEvent = 'click'
+
+const shownNotifications = new LRUCache<string, DesktopNotification>({
+  max: 50,
+})
+
+function onNotificationEvent(event: DesktopNotificationEvent, id: string) {
+  const notification = shownNotifications.get(id)
+  if (notification === undefined) {
+    // TODO: handle notifications that are not in the cache
+    return
+  }
+
+  if (event === 'click') {
```

Comment:
So we can't even be sure that the notification is dismissed if it's clicked? Can it still live on in the notification center?

Reply:
We can, yeah! Before this PR, I had some SEGFAULT crashes trying to remove the notification and its callback from memory right after the `click` event. Then I forgot to do that. I'll tackle that in a different PR along with other minor improvements I want to do.